FEM-2405 : Full screen/Zoom support devices specifically for 18:9 devices to prevent letter boxing of video playback

amitsamal opened the pull request:
Feature change - Full screen/Zoom support devices specifically for 18:9 devices to prevent letter boxing of video playback
Kaltura internal ticket - https://kaltura.atlassian.net/browse/VIA2-358

### Description of the Changes

1. Added changes to contentFrame in order to handle changes to resizeMode of AspectRatioFrameLayout of exoplayerview. This would help us add zoom mode functionality and prevent letterboxing to videos on 18:9 devices. 
2. Add API to player Settings for start Mode:
```
player.getSettings().setSurfaceAspectRatioResizeMode(PKAspectRatioResizeMode.fill);
```
3. API added tp `Player` object for dynamic mode change during playback 
```
player.updateSurfaceAspectRatioResizeMode(PKAspectRatioResizeMode resizeMode) in 
```
4. add new event ASPECT_RATIO_RESIZE_MODE_CHANGED

### enum
public enum PKAspectRatioResizeMode {

    fit,
    fixedWidth,
    fixedHeight,
    fill,
    zoom
}
```

Commit:
Feature : Added control to zoom surface view for fullscreen support on devices specifically for 18:9 devices to prevent letter boxing of video playback

Commit:
Added check to add only permitted values for aspect ratio change. Code refractor.

amitsamal committed:
Merge branch 'dev' into dev

kaltura-hooks commented:
Hi @amitsamal,
Thank you for contributing this pull request!
Please sign the Kaltura CLA so we can review and merge your contribution.
Learn more at http://bit.ly/KalturaContrib

Commit:
Missed Annotation for preventing

Commit:
Merge branch 'dev' of https://github.com/amitsamal/playkit-android into dev

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

giladna committed:
use enum for resize mode instead of AspectRatioFrameLayout.ResizeMode

giladna committed:
remove - already defined in BasePlayerView

giladna committed:
code review changes

giladna committed:
rename to zoom

giladna committed:
Update PlayerSettings.java

rename to getAspectRatioResizeMode

giladna committed:
Update ExoPlayerWrapper.java

rename to getAspectRatioResizeMode

giladna committed:
Update ExoPlayerWrapper.java

giladna committed:
Update PlayerController.java

getAspectRatioResizeMode

giladna committed:
Update PlayerEngine.java

cosmetics

giladna committed:
Update ExoPlayerWrapper.java

rename to configureAspectRatioResizeMode

giladna committed:
Update ExoPlayerView.java

remove space

giladna committed:
Update PlayerController.java

fix log updateSurfaceAspectRatioResizeMode

giladna committed:
Update PlayerController.java

update method name updateSurfaceAspectRatioResizeMode

giladna committed:
Update Player.java

update comment

giladna committed:
Update PlayerEngine.java

update comment

giladna committed:
Update ExoPlayerView.java

updating the view as soon as update was called

        if (contentFrame != null) {
            contentFrame.setResizeMode(this.resizeMode);
        }

noamtamim committed:
Update PKAspectRatioResizeMode.java

giladna committed:
Update PlayerEngine.java

add @param
